src/components/Survey/Nickname/Nickname.tsx

```diff
@@ -12,60 +11,85 @@ const {
   s_title,
   s_errorMsg,
   s_errorIcon,
-} = styles;
+  s_nextButton,
+} = style;
 
 interface Props {
   nickname: string;
   setNickname: (value: string) => void;
   setNextPage: () => void;
 }
 
-function Nickname({ nickname, setNickname, setNextPage }: Props) {
-  const [isDisabled, setIsDisabled] = useState<boolean>(!nickname);
+const Nickname = ({ nickname, setNickname, setNextPage }: Props) => {
   const [errorMessage, setErrorMessage] = useState<string>('');
+  const isDisabled = useMemo(() => !nickname || !!errorMessage, [nickname, errorMessage]);
 
-  const checkNicknameValidation = (value: string) => {
-    const nicknameRegex = /^[가-힣\s|ㄱ-ㅎ|a-z|A-Z|0-9|_|.|,]+$/g;
-    const nicknameLengthRegex = /^.{2,8}$/g;
+  useEffect(() => {
+    const isValidNickname = (userNickname: string) => {
+      const nicknameSpaceRegex = /\s/g;
+      const specialCharactersRegex = /[`~!@#$%^&*()|+\-=?;:'"<>\\{\\}\\[\]\\\\/]/g;
+      const nicknameRegex = /^[가-힣\s|ㄱ-ㅎ|a-z|A-Z|0-9|_|.|,]+$/g;
+      const nicknameLengthRegex = /^.{2,8}$/g;
 
-    if (!nicknameRegex.test(value)) {
-      setErrorMessage('한글, 영어, 숫자, 특수문자(_.,)만 가능해요');
-      setIsDisabled(true);
-    } else if (!nicknameLengthRegex.test(value)) {
-      setErrorMessage('2 ~ 8글자만 가능해요');
-      setIsDisabled(true);
+      if (nicknameSpaceRegex.test(userNickname)) {
+        return '공백을 제거해 주세요.';
+      }
+      if (specialCharactersRegex.test(nickname)) {
+        return '특수문자는 . , _ 만 가능해요';
+      }
+
+      if (!nicknameRegex.test(userNickname)) {
+        return '한글, 영어, 숫자, 특수문자(_.,)만 가능해요';
+      }
+      if (!nicknameLengthRegex.test(userNickname)) {
+        return '2 ~ 8글자만 가능해요';
+      }
+
+      return null;
+    };
+
+    const nickNameError = isValidNickname(nickname);
+
+    if (nickNameError) {
+      setErrorMessage(nickNameError);
     } else {
       setErrorMessage('');
-      setIsDisabled(false);
     }
-  };
+  }, [nickname]);
 
-  // onChange
   const handleChangeNickname = (e: React.ChangeEvent<HTMLInputElement>): void => {
     const { value }: { value: string } = e.target;
     setNickname(value);
-    checkNicknameValidation(value);
   };
 
   return (
     <section className={classNames(s_container)}>
-      <h2 className={classNames(s_title)}>당신을 뭐라고 불러드릴까요?</h2>
+      <h2 className={classNames(s_title)}>
+        <p>당신을</p> 뭐라고 불러드릴까요?
```

Comment:
줄바꿈 스타일링 때문에 p태그를 사용한거 같은데 시멘틱한 관점에서 봤을때 당신을 ~ 불러드릴까요? 까지가 p태그의 자식으로 와야할것 가고 당신을은 스타일을 위해 태그로 감싸준것이니 큰 의미가 없다면 span을 사용해주는게 더 좋은거 같아요!
그리고 줄바꿈을 위해 쓰는 스타일링은 common.scss에 s_whiteSpace라는 속성이 정의되어 있어서 바로 사용하면 됩니다!
```suggestion
        <p className={classNames(s_title)}>
          <span className={classNames('s_whiteSpace')}>당신을</span>뭐라고 불러드릴까요?
        </p>
```